boot/zephyr/prj.conf

```diff
@@ -1,5 +1,6 @@
 CONFIG_CONSOLE_HANDLER=y
-CONFIG_SYS_LOG=y
+CONFIG_SYS_LOG=n
```

Comment:
I think as a matter of design:

- if zephyr's sys_log logging is enabled, its mcuboot_config.h should automatically enable MCUboot's logs
- if logging is disabled, its mcuboot_config.h should make sure MCUboot's logs are disabled

That way there is no need for an additional BOOT_HAVE_LOGGING option, which can then simply be deleted. MCUboot logging will "just work" based on Zephyr logging using the configuration here, which users can override using the config options they are already used to. Zephyr logging is pretty confusing, I think, so adding extra options should be avoided if possible.

If desired, this series could instead add a `config BOOT_DEFAULT_LOG_LEVEL` which would control the MCUboot-specific default log level, leaving the syslog default level alone for other Zephyr subsystems used by MCUboot which may be affected by the default.